### PR TITLE
add options to delay broadcast notifications

### PR DESCRIPTION
**Pull request by ice-cronus**

### CODE REVIEW

File: notifications/push/contract.go [hunk 1]
```diff
@@ -47,6 +49,15 @@ const (
 	requestDeadline                                  = 25 * stdlibtime.Second
 	fcmSendAllBufferingDeadline                      = 1 * stdlibtime.Second
 	fcmSendAllSlowProcessingMonitoringTickerDeadline = 3 * fcmSendAllBufferingDeadline
+
+	dataOnlyTitle    = "title"
+	dataOnlyBody     = "body"
+	dataOnlyImageURL = "imageURL"
```
Comment (ice-ares) [inline notifications/push/contract.go]: imageUrl

File: notifications/push/push.go [hunk 2]
```diff
@@ -124,19 +125,51 @@ func (p *push) Send(ctx context.Context, notif *Notification[DeviceToken], respo
...
+			Android: buildAndroidDataOnlyNotification(notification),
+			APNS:    buildAppleNotification(notification),
+			Topic:   string(notification.Target),
 		})
 
 		return err //nolint:wrapcheck // No need to do that, it's wrapped outside.
 	}), "[%v] permanently failed to broadcast %#v", p.applicationYAMLKey, notification)
 }
 
+func buildAndroidDataOnlyNotification(notification *Notification[SubscriptionTopic]) *fcm.AndroidConfig {
```
Comment (ice-ares) [inline notifications/push/push.go]: No, u don't control this here, I said we send to both topics (delayed and regular), so u have to support that in the api, modify the contract

File: notifications/push/push.go [hunk 3]
```diff
@@ -124,19 +125,51 @@ func (p *push) Send(ctx context.Context, notif *Notification[DeviceToken], respo
...
+	dataOnlyNotification[dataOnlyMaxDelay] = strconv.FormatUint(uint64(notification.MaxDelay), 10)
+
+	return &fcm.AndroidConfig{
+		Data:     dataOnlyNotification,
+		Priority: "high",
+	}
+}
+
+func buildAppleNotification(notification *Notification[SubscriptionTopic]) *fcm.APNSConfig {
+	return &fcm.APNSConfig{
```
Comment (ice-ares) [inline notifications/push/push.go:173]: u don't set data here?

Comment (ice-cronus) [reply notifications/push/push.go:173]: It uses data from notification.Data, android overwrites tgem as it needs extra fields in it

File: notifications/push/contract.go [hunk 4]
```diff
@@ -31,11 +31,17 @@ type (
 		Body     string `json:"body,omitempty"`
 		ImageURL string `json:"imageUrl,omitempty"`
 	}
+	DelayedNotification struct {
+		*Notification[SubscriptionTopic]
+		MinDelay uint `json:"minDelay"`
```
Comment (ice-ares) [inline notifications/push/contract.go]: MinDelaySec uint `json:"minDelaySec"`

File: notifications/push/contract.go [hunk 5]
```diff
@@ -31,11 +31,17 @@ type (
...
+		MaxDelay uint `json:"maxDelay"`
```
Comment (ice-ares) [inline notifications/push/contract.go]: MaxDelaySec uint `json:"maxDelaySec"`

**Pull request merged**